src/ui/public/chrome/directives/app_switcher/app_switcher.js

```diff
@@ -1,10 +1,18 @@
 import DomLocationProvider from 'ui/dom_location';
 import { parse } from 'url';
-import { bindKey } from 'lodash';
+import { bindKey, filter } from 'lodash';
 import '../app_switcher/app_switcher.less';
 import uiModules from 'ui/modules';
 import appSwitcherTemplate from './app_switcher.html';
 
+function isNavLinkEnabled(link) {
+  return link.enable === undefined || link.enable;
+}
+
+function isNavLinkShown(link) {
+  return link.show === undefined || link.show;
```

Comment:
Okay, bringing our out-of-band discussion into this thread + considering what @cjcenizal just posted in the previous comment, here's what I'm going to do for this PR:

* Define the properties explicitly in the class as `hidden` and `disabled`.
* Default both of them to `false`.
* Keep the names of the `isNavLinkShown` and `isNavLinkEnabled` functions in `app_switcher.js` as-is but update the logic within them to use the `hidden` and `disabled` properties respectively.